File: examples/time/time_range.py
# /// script
# dependencies = ["brahe"]
# ///
"""
Initialize EOP Providers with simpliest way possible
"""

import brahe as bh

bh.initialize_eop()

for epc in bh.TimeRange(
    bh.Epoch(2024, 1, 1, 0, 0, 0.0, time_system=bh.UTC),
    bh.Epoch(2024, 1, 2, 0, 0, 0.0, time_system=bh.UTC),
    3600.0,
):
    print(epc)

File: examples/time/time_range.rs
//! Initialize EOP Providers with simpliest way possible

#[allow(unused_imports)]
use brahe as bh;

fn main() {
    bh::initialize_eop().unwrap();

    for epc in bh::TimeRange::new(
        bh::Epoch::from_datetime(2024, 1, 1, 0, 0, 0.0, 0.0, bh::TimeSystem::UTC),
        bh::Epoch::from_datetime(2024, 1, 2, 0, 0, 0.0, 0.0, bh::TimeSystem::UTC),
        3600.0
    ) {
        println!("{}", epc);
    }
}
